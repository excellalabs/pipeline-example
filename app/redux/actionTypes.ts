
export const setName = 'SET_NAME'
