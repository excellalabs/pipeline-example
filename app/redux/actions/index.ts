
export { default as SetName } from './setName'
