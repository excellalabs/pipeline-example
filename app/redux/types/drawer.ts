
export interface Drawer {
  open: boolean
}
