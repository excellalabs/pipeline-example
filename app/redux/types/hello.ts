
export interface Hello {
  name: string
}
