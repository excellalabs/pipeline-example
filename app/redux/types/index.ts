
export { State } from './state'
export { Hello } from './hello'
