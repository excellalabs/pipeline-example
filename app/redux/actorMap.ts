
import { Action, Reducer } from 'redux'

export interface ActorMap<S> {
  [name: string]: Reducer<S>
}

export function buildReducer<S> (initialState: S, map: ActorMap<S>): Reducer<S> {
  return (prev: S = initialState, action: Action): S => {
    let actor = map[action.type]
    return typeof actor === 'function' ? actor(prev, action) : prev
  }
}
