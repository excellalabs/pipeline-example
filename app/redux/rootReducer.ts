
import { Reducer, Action, combineReducers } from 'redux'
import { routerReducer } from 'react-router-redux'

import { State } from './types'
import helloReducer, { INITIAL_STATE as HELLO } from './reducers/hello'

const INITIAL_STATE: State = {
  hello: HELLO
}

export default combineReducers<State>({
  hello: helloReducer,
  routing: routerReducer
})
