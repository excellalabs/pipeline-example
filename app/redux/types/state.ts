
import { Hello } from './hello'

export interface State {
  hello: Hello
}
