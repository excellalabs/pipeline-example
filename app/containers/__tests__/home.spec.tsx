
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import Home from '../home'
import { State } from '../../redux/types'

const state: State = {
  drawer: {
    open: false
  }
}

const store = {
  getState: jest.fn(() => state),
  dispatch: jest.fn(),
  subscribe: jest.fn(),
  replaceReducer: jest.fn()
}

const setName = (name: string) => {
  //state.hello.name = name
}

xdescribe('snapshots', () => {
  it('should display the correct default name', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Home />
      </Provider>
    )
    expect(tree).toMatchSnapshot()
  })

  it('should display the correct name when it is set', () => {
    setName('John Doe')
    const tree = renderer.create(
      <Provider store={store}>
        <Home />
      </Provider>
    )
    expect(tree).toMatchSnapshot()
  })
})
