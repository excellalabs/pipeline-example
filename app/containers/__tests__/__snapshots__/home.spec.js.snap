exports[`snapshots should display the correct default name 1`] = `
<div>
  <div>
    <div>
      Hello, 
      world
    </div>
    <input
      onChange={[Function]}
      type="text"
      value="world" />
  </div>
</div>
`;

exports[`snapshots should display the correct name when it is set 1`] = `
<div>
  <div>
    <div>
      Hello, 
      John Doe
    </div>
    <input
      onChange={[Function]}
      type="text"
      value="John Doe" />
  </div>
</div>
`;
